src/components/common/BtnTask/BtnTask.tsx

```diff
@@ -0,0 +1,214 @@
+import { css, Theme } from '@emotion/react';
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+
+import BtnDate from '../BtnDate/BtnDate';
+import StatusDoneBtn from '../button/statusBtn/StatusDoneBtn';
+import StatusInProgressBtn from '../button/statusBtn/StatusInProgressBtn';
+import StatusStagingBtn from '../button/statusBtn/StatusStagingBtn';
+import StatusTodoBtn from '../button/statusBtn/StatusTodoBtn';
+
+import Icons from '@/assets/svg/index';
+
+interface BtnTaskProps {
+	btnType: number;
+	status?: string;
+	isDescription?: boolean;
+}
+
+interface BorderColorProps {
+	isHovered: boolean;
+	isClicked: boolean;
+	iconHovered: boolean;
+	theme: Theme;
+	btnType: number;
+}
+
+function BtnTask(props: BtnTaskProps) {
+	const { btnType, status = 'Todo', isDescription = false } = props;
+	const [isClicked, setIsClicked] = useState(false);
+	const [isHovered, setIsHovered] = useState(false);
+	const [iconHovered, setIconHovered] = useState(false);
+	const [iconClicked, setIconClicked] = useState(false);
+
+	const handleMouseEnter = () => {
+		setIsHovered(true);
+	};
+
+	const handleMouseLeave = () => {
+		setIsHovered(false);
+	};
+
+	const handleClick = () => {
+		setIsClicked((prev) => !prev);
+	};
+
+	const stopPropagation = (e: React.MouseEvent) => {
+		e.stopPropagation();
+	};
+
+	const handleIconMouseEnter = () => {
+		setIconHovered(true);
+	};
+
+	const handleIconMouseLeave = () => {
+		setIconHovered(false);
+	};
+
+	const handleIconClick = (e: React.MouseEvent) => {
+		if (iconClicked) {
+			setIconClicked(false);
+		} else {
+			stopPropagation(e);
+			setIconClicked(true);
+		}
+	};
+
+	const renderStatusButton = () => {
+		if (btnType === 3) {
+			return <StagingIconHoverIndicator />;
+		}
+
+		if (!iconHovered && !iconClicked) {
+			return <IconHoverIndicator />;
+		}
+
+		if (btnType === 1) {
+			return <StatusStagingBtn />;
+		}
+
+		if (btnType === 2) {
+			if (status === 'Done') {
+				return <StatusDoneBtn />;
+			}
+			if (status === 'InProgress') {
+				return <StatusInProgressBtn />;
+			}
+			return <StatusTodoBtn />;
+		}
```

Comment:
```suggestion
	switch (status) {
            case 'Done':
                return <StatusDoneBtn />;
            case 'InProgress':
                return <StatusInProgressBtn />;
            default:
                return <StatusTodoBtn />;
        }
```

이미 간단하긴 하지만 switch 문 사용해서 더 명확하게 표현할 수 있을 것 가타요